base/src/amgx_cusparse.cu

```diff
@@ -1074,7 +1123,11 @@ inline void Cusparse::bsrmv( cusparseHandle_t handle, cusparseDirection_t dir, c
 {
     if (blockDim == 1)
     {
-        cusparseCheckError(cusparseScsrmv(handle, trans, mb, nb, nnzb, alpha, descr, bsrVal, bsrRowPtr, bsrColInd, x, beta, y));
+        #ifdef CUSPARSE_GENERIC_INTERFACES
+            generic_SpMV(handle, trans, mb, nb, nnzb, alpha, bsrVal, bsrRowPtr, bsrColInd, x, beta, y, CUDA_R_32F, CUDA_R_32F);
```

Comment:
It appears to be that we are loosing `IndexType` here. Cusparse::bsrmv() has always `int` for indices type, even though TConfig can have int64. And inside generic_SpMV we hardcode `CUSPARSE_INDEX_32I`. Currently int64 is not used as a option for indices type so it doesn't complain. But when someone will try using int64 for matrix indices - it will require much more code changes.

Reply:
Yes I think we will need a whole additional set of overloads to account for that. In the (very) long term we will be able to completely remove any reference to the non-generic cuSPARSE routines and that way we should be able to actually take more advantage of the generic interfaces and reduce the overall footprint of this code.